### PR TITLE
[IDGEN-78] Link Log Entries EndPoints to OWA

### PR DESCRIPTION
**Pull request by cnagadya**
## JIRA TICKET NAME:

[IDGEN-78 Link Log Entries EndPoints to OWA](https://issues.openmrs.org/browse/IDGEN-78)

## SUMMARY:
Link Log Entries EndPoints to OWA

### CODE REVIEW

File: owa/app/js/components/view_log_entries/viewLogEntries.jsx [hunk 1]
```diff
@@ -9,127 +9,284 @@
...
+        this.getIdentifierSources = this.getIdentifierSources.bind(this);
+        this.advancedSearchView = this.advancedSearchView.bind(this);
+        this.advancedSearch = this.advancedSearch.bind(this);
+        this.handleSearch = this.handleSearch.bind(this);
+        this.handleChangeStart = this.handleChangeStart.bind(this);
+        this.handleChangeEnd = this.handleChangeEnd.bind(this);
+        this.resetSearchForm=this.resetSearchForm.bind(this);
+    }
+    getLogEntries() {
+        apiCall(null, 'get', '/idgen/logentry?v=full&identifier=&comment=&source&generatedBy=&dateGenerated=').then((response) => {
```
Comment (Ewanjiru) [inline owa/app/js/components/view_log_entries/viewLogEntries.jsx]: This line seems to be way too long than the required 80 characters..

File: owa/app/js/components/view_log_entries/viewLogEntries.jsx [hunk 2]
```diff
@@ -9,127 +9,284 @@
...
             isHidden: !this.state.isHidden
         })
     }
+
+    advancedSearch() {
+        const {source, identifier, comment} = this.state.searchFilters;
+        const endDate = this.state.endDate.toISOString();
+        const generatedBy = this.state.generatedBy;
+        if (!this.state.startDate){
+            apiCall(null, 'get', `/idgen/logentry?v=full&generatedBy=${generatedBy}&identifier=${identifier}&comment=${comment}&source=${source}&toDate=${endDate}`).then((response) => {
```
Comment (Ewanjiru) [inline owa/app/js/components/view_log_entries/viewLogEntries.jsx]: Ditto

File: owa/app/js/components/view_log_entries/viewLogEntries.jsx [hunk 3]
```diff
@@ -9,127 +9,284 @@
...
+            this.setState({logEntries: response.results});
+        });
+        }
+        else{ 
+                const startDate = this.state.startDate.toISOString();
+                apiCall(null, 'get', `/idgen/logentry?v=full&identifier=${identifier}&comment=${comment}&source=${source}&generatedBy=${generatedBy}&fromDate=${startDate}&toDate=${endDate}`).then((response) => {
```
Comment (Ewanjiru) [inline owa/app/js/components/view_log_entries/viewLogEntries.jsx]: Ditto

File: owa/app/js/components/view_log_entries/viewLogEntries.jsx [hunk 4]
```diff
@@ -9,127 +9,284 @@
...
+                    this.setState({logEntries: response.results});
+                });
+        }
+        this.setState({
+            isHidden: !this.state.isHidden
+        })
+    }
```
Comment (Ewanjiru) [inline owa/app/js/components/view_log_entries/viewLogEntries.jsx]: separate blocks with a single line

File: owa/app/js/components/view_log_entries/viewLogEntries.jsx [hunk 5]
```diff
@@ -9,127 +9,284 @@
...
+                    this.setState({ generatedBy: value});
+                }
+            }
+            
+        }
+        else{this.setState({
+            searchFilters: Object.assign({}, this.state.searchFilters, {[name]: value})
+        });
+    }
+    }
```
Comment (Ewanjiru) [inline owa/app/js/components/view_log_entries/viewLogEntries.jsx]: Separate blocks with single line

**Comment by cnagadya** [comment]
Changes have been made @Ewanjiru  :)

File: owa/app/js/components/view_log_entries/viewLogEntries.jsx [hunk 6]
```diff
@@ -9,127 +9,284 @@
...
+                                        value="Search"
+                                        className=" col-sm-6 button confirm"
+                                        onClick={this.advancedSearch}/>
+                                        <input type="button" className="col-sm-6 cancelbtn" value="Reset" onClick={this.resetSearchForm}/>
+
+                                </div>
+                            </form>
+                        </div>}</div>
                     <div className="table-responsive logs_table">
+                        { (this.state.searchFilters.source || this.state.searchFilters.identifier || this.state.generatedByValue || this.state.searchFilters.comment || this.state.startDate|| this.state.dateChanged)  &&
```
Comment (Ewanjiru) [inline owa/app/js/components/view_log_entries/viewLogEntries.jsx]: Line too long

Comment (Ewanjiru) [reply owa/app/js/components/view_log_entries/viewLogEntries.jsx]: Your whole file seems to have more than 2spaces of indentation, look into that.

Comment (cnagadya) [reply owa/app/js/components/view_log_entries/viewLogEntries.jsx]: Was using 4 spaces probably because that's what I used to do for Python projects but just searched online & I've noticed most JS developers are pro soft indentation so I've reduced them to 2.

File: owa/app/js/components/view_log_entries/viewLogEntries.jsx [hunk 7]
```diff
@@ -9,127 +9,308 @@
...
+      this.setState({identifierSources: response.results});
+    });
+  }
 
+  advancedSearchView() {
+    this.setState({
+      isHidden: !this.state.isHidden
+    })
+  }
+
```
Comment (jacksono) [inline owa/app/js/components/view_log_entries/viewLogEntries.jsx]: Since you are updating the 'isHidden' state using a modification of its current value, you might want to consider using  the "prevState" construct. This will avoid any potential race-conditions that arise as a result of the setState method being an asynchronous one.

File: owa/app/js/components/view_log_entries/viewLogEntries.jsx [hunk 8]
```diff
@@ -9,127 +9,308 @@
...
-                    </div>
-                    <button
-                        className="searchlink"
-                        onClick={this
-                        .advancedSearchView
-                        .bind(this)}>More Search Options</button>
+    this.setState({
+      isHidden: !this.state.isHidden
+    });
+  }
```
Comment (jacksono) [inline owa/app/js/components/view_log_entries/viewLogEntries.jsx]: Ditto

File: owa/app/js/components/view_log_entries/viewLogEntries.jsx [hunk 9]
```diff
@@ -9,127 +9,308 @@
...
+          </div>
+          <button
+            className="searchlink"
+            onClick={this.advancedSearchView}>More Search Options</button>
+        </div>
+        <div className="logs_table_area">
+          <div >{!this.state.isHidden && <div className="advanced_wrapper">
+            <form id="advancedSearchForm">
+              <br/>
+              <fieldset>
```
Comment (jacksono) [inline owa/app/js/components/view_log_entries/viewLogEntries.jsx]: Having 2 div tags on the same line is confusing to read, consider using indentations to separate the elements hierarchically

**Comment by cnagadya** [comment]
Thanks @jacksono. Recommended changes have been incorporated :)

File: omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java [hunk 10]
```diff
@@ -98,7 +98,7 @@ public DelegatingResourceDescription getRepresentationDescription(Representation
 
 	@PropertyGetter("source")
 	public String getSource(LogEntry logentry) {
-		return logentry.getSource().getUuid();
+		return logentry.getSource().getName();
```
Comment (kwahalf) [inline omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java]: why did you change this?

**Comment by cnagadya** [comment]
@dkayiwa, kindly review this PR.

File: omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java (hunk 10 above)
Comment (cnagadya) [reply omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java]: For usability, the Identifier Source names should be displayed and not the uuids

Comment (kwahalf) [reply omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java]: if you remove the property getter for source, since you don't need it, the source property will return the whole source object that contains all the source fields such as name, Uuid etc

Comment (cnagadya) [reply omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java]: I need to return just the names because I passing them as options to a select

Comment (kwahalf) [reply omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java]: @dkayiwa what's your take on the above?

Comment (dkayiwa) [reply omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java]: Completely remove this proper PropertyGetter.
We use them for properties that are not part of the resource object. For instance LogEntry does not have a display property and hence why we need a PropertyGetter for it. But for source, we do not need it. Therefore remove it completely.

**Comment by cnagadya** [comment]
Fixed

File: owa/app/css/openmrs-owa-idgen.css [hunk 11]
```diff
@@ -17,13 +17,16 @@ body {
     font-size: 16px;
 }
 
+html body {
```
Comment (kwahalf) [inline owa/app/css/openmrs-owa-idgen.css]: why was this added?

Comment (cnagadya) [reply owa/app/css/openmrs-owa-idgen.css]: obviously to center the body regardless of screen size

Comment (kwahalf) [reply owa/app/css/openmrs-owa-idgen.css]: have you checked the effects it has on the other tabs apart from yours

Comment (kwahalf) [reply owa/app/css/openmrs-owa-idgen.css]: screen shot 2017-10-29 at 10 11 48

Comment (kwahalf) [reply owa/app/css/openmrs-owa-idgen.css]: this is how it looks in full screen. please remove it

Comment (cnagadya) [reply owa/app/css/openmrs-owa-idgen.css]: if you noticed, I removed the left: 140px property that you had put on most elements in the layout. This is because it rendered the page contents 140px away from the left creating the illusion that it is centred on all devices with screen sizes like ours but in actuality it wasnt

Comment (kwahalf) [reply owa/app/css/openmrs-owa-idgen.css]: Yes seen it  looks good

Comment (kwahalf) [reply owa/app/css/openmrs-owa-idgen.css]: also don't forget to remove the left: 140px in breadcrumbs.css

Comment (cnagadya) [reply owa/app/css/openmrs-owa-idgen.css]: I already removed it and committed the changes.

Comment (kwahalf) [reply owa/app/css/openmrs-owa-idgen.css]: All good @cnagadya

File: omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java [hunk 12]
```diff
@@ -96,11 +96,6 @@ public DelegatingResourceDescription getRepresentationDescription(Representation
 		return description;
 	}
 
-	@PropertyGetter("source")
-	public String getSource(LogEntry logentry) {
```
Comment (dkayiwa) [inline omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java]: Shouldn't this be in a separate pull request as a followup commit for IDGEN-67

Comment (cnagadya) [reply omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java]: Fixed
